background_scripts/bg_utils.js

```diff
@@ -14,6 +14,41 @@ const BgUtils = {
       : false;
   },
 
+  // Get a query dictionary for `chrome.tabs.query` that will only return the visible tabs.
+  visibleTabs() {
+    const visibleTabsQuery = {
+      currentWindow: true,
+    };
+    // Only Firefox supports hidden tabs
+    if (this.isFirefox()) {
+      visibleTabsQuery.hidden = false;
+    }
+    return visibleTabsQuery;
+  },
+
+  // Find the real tab index in a given tab array.
+  tabIndex(tab, tabs) {
```

Comment:
We should document why we need to do this. E.g. "in the case of hidden tabs in Firefox browsers, tab.index may not correspond to the actual index in the visible tabs returned by tabs.query." Otherwise, in the code, it's pretty implicit that our tab logic needs to handle hidden tabs, and this will probably get regressed over time. Let's note it explicitly.

Also can callers use `tabs.findIndex` (or `Array.from(tabs).findIndex` if necessary, although that incurs a full array copy) instead of having this separate utility method?